transforms/universal/html2parquet/python/src/html2parquet_transform.py

```diff
@@ -0,0 +1,117 @@
+import time
+from argparse import ArgumentParser, Namespace
+from typing import Any
+import zipfile
+import io
+import trafilatura
+from datetime import datetime
+
+import pyarrow as pa
+
+# disabled for now
+# from data_processing_ray.runtime.ray import RayTransformLauncher
+# from data_processing_ray.runtime.ray.runtime_configuration import (
+#   RayTransformRuntimeConfiguration,
+# )
+# import data_processing
+
+
+from data_processing.transform import AbstractBinaryTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider, get_logger, TransformUtils
+
+
+class Html2ParquetTransform(AbstractBinaryTransform):
+    def __init__(self, config: dict[str, Any]):
+        super().__init__(config)  
+
+    def _convert_html2parquet(self, member_filename:str, file_name:str, content_bytes: bytes) -> dict:
+        title = member_filename if member_filename else TransformUtils.get_file_basename(file_name)
+        
+        # Use Trafilatura library
+        content_string = trafilatura.extract(content_bytes)
+
+        if content_string is None:
+            raise RuntimeError("Failed in converting.")
+
+        row_data = {
+            "title": title,
+            "document": TransformUtils.get_file_basename(file_name),
+            "contents": content_string,
+            "document_id": TransformUtils.str_to_hash(content_string),
+            "size": len(content_string),
+            "date_acquired": datetime.now().isoformat()
+        }
+
+        return row_data
+
+    def transform_binary(self, file_name: str, byte_array: bytes) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        Converts raw data file (ZIP) / raw HTMLs to Parquet format
+
+        If file_name is detected as a HTML file, it generates a pyarrow table with a single row
+        that contains the document converted to a text string.
+        If file_name is detected as a ZIP archive, it generates a pyarrow table with a row
+        for each HTML file detected in the archive.
+        """
+        if TransformUtils.get_file_extension(file_name)[1] not in [".zip", ".html"]:
+            error_message = f"Unsupported file type: {file_name}. Only ZIP and HTML files are supported."
+            self.logger.error(error_message)
+            raise ValueError(error_message)  # Raising an exception with the error message
+        data = []
+        number_of_rows = 0
+
+        # Process ZIP archive of HTML documents
+        if(TransformUtils.get_file_extension(file_name)[1] == ".zip"):
+            with zipfile.ZipFile(io.BytesIO(bytes(byte_array))) as opened_zip:
+                # Loop through each file member in the ZIP archive
+                for member in opened_zip.infolist():
+                    if not member.is_dir() and '__MACOSX' not in member.filename and TransformUtils.get_file_extension(member.filename)[1] == ".html":
```

Comment:
why would you need to MACOSX test if you're always requiring .html extension?

Reply:
You are right. This MACOSX test was added before requiring .html extension. Deleted it.